File: CS/10_StampsAndWatermarks/ImageAndTextUsingTemplate.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Imaging;
using System.IO;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;

namespace ImageAndTextUsingTemplate
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            //Load Pdf document from disk
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile("../../../../../../../Data/PDFTemplate_HF.pdf");

            //Get the first page
            PdfPageBase page = doc.Pages[0];

            //Get the margins of Pdf
            PdfMargins margin = doc.PageSettings.Margins;

            //Define font and brush
            PdfTrueTypeFont font=new PdfTrueTypeFont(new Font("Impact",14f,FontStyle.Regular));
            PdfSolidBrush brush=new PdfSolidBrush(Color.Gray);

            //Load an image
            PdfImage image = PdfImage.FromFile("../../../../../../../Data/E-iceblueLogo.png");

            //Specify the image size
            SizeF imageSize=new SizeF(image.Width/2,image.Height/2);

            //Create a header template
            PdfTemplate headerTemplate = new PdfTemplate(page.ActualSize.Width - margin.Left - margin.Right, imageSize.Height);

            //Draw the image in the template
            headerTemplate.Graphics.DrawImage(image, new PointF(0, 0),imageSize);

            //Create a retangle
            RectangleF rect = headerTemplate.GetBounds();

            //string format
            PdfStringFormat format1=new PdfStringFormat(PdfTextAlignment.Right,PdfVerticalAlignment.Middle);

            //Draw a string in the template
            headerTemplate.Graphics.DrawString("Header", font, brush, rect, format1);

            //Create a footer template and draw a text
            PdfTemplate footerTemplate = new PdfTemplate(page.ActualSize.Width - margin.Left - margin.Right, imageSize.Height);
            PdfStringFormat format2=new PdfStringFormat(PdfTextAlignment.Center,PdfVerticalAlignment.Middle);
            footerTemplate.Graphics.DrawString("Footer", font, brush, rect, format2);

            float x = margin.Left;
            float y = 0;

            //Draw the header template on page at specified location
            page.Canvas.DrawTemplate(headerTemplate, new PointF(x,y));

            //Draw the footer template on page at specified location
            y = page.ActualSize.Height - footerTemplate.Height - 10;
            page.Canvas.DrawTemplate(footerTemplate, new PointF(x, y));

            //Save the document
            string output = "ImageAndTextInHeaderFooterUsingTemplate_out.pdf";
            doc.SaveToFile(output, FileFormat.PDF);

            //Launch the Pdf document
            PDFDocumentViewer(output);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/10_StampsAndWatermarks/ImageAndTextUsingTemplate.vb
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace ImageAndTextUsingTemplate
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new PdfDocument object
			Dim doc As New PdfDocument()

			' Load the PDF document from the specified file
			doc.LoadFromFile("../../../../../../../Data/PDFTemplate_HF.pdf")

			' Get the first page of the document
			Dim page As PdfPageBase = doc.Pages(0)

			' Get the margins of the document's page settings
			Dim margin As PdfMargins = doc.PageSettings.Margins

			' Create a TrueType font object with Impact font, size 14, and regular style
			Dim font As New PdfTrueTypeFont(New Font("Impact", 14.0F, FontStyle.Regular))
			' =============================================================================
			' Use the following code for netstandard dlls
			' =============================================================================
			'Dim font As New PdfTrueTypeFont("Impact", 14.0F, FontStyle.Regular, True)
			' =============================================================================

			' Create a solid brush with gray color
			Dim brush As New PdfSolidBrush(Color.Gray)

			' Load the image for the header/footer section
			Dim image As PdfImage = PdfImage.FromFile("../../../../../../../Data/E-iceblueLogo.png")

			' Calculate the size of the image
			Dim imageSize As New SizeF(image.Width \ 2, image.Height \ 2)

			' Create a new PdfTemplate object for the header section
			Dim headerTemplate As New PdfTemplate(page.ActualSize.Width - margin.Left - margin.Right, imageSize.Height)

			' Draw the image on the header template
			headerTemplate.Graphics.DrawImage(image, New PointF(0, 0), imageSize)

			' Get the bounds of the header template
			Dim rect As RectangleF = headerTemplate.GetBounds()

			' Create a string format for aligning the text in the header template (right-aligned)
			Dim format1 As New PdfStringFormat(PdfTextAlignment.Right, PdfVerticalAlignment.Middle)

			' Draw the header text on the header template
			headerTemplate.Graphics.DrawString("Header", font, brush, rect, format1)

			' Create a new PdfTemplate object for the footer section
			Dim footerTemplate As New PdfTemplate(page.ActualSize.Width - margin.Left - margin.Right, imageSize.Height)

			' Create a string format for aligning the text in the footer template (center-aligned)
			Dim format2 As New PdfStringFormat(PdfTextAlignment.Center, PdfVerticalAlignment.Middle)

			' Draw the footer text on the footer template
			footerTemplate.Graphics.DrawString("Footer", font, brush, rect, format2)

			' Define initial coordinates for positioning the header/footer templates on the page canvas
			Dim x As Single = margin.Left
			Dim y As Single = 0

			' Draw the header template on the page canvas
			page.Canvas.DrawTemplate(headerTemplate, New PointF(x, y))

			' Update the y-coordinate for positioning the footer template
			y = page.ActualSize.Height - footerTemplate.Height - 10

			' Draw the footer template on the page canvas
			page.Canvas.DrawTemplate(footerTemplate, New PointF(x, y))

			' Save the modified PDF document to a file
			Dim output As String = "ImageAndTextInHeaderFooterUsingTemplate_out.pdf"
			doc.SaveToFile(output, FileFormat.PDF)

			' Close the document
			doc.Close()

			' Launch the Pdf document
			PDFDocumentViewer(output)
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
